bin/halcyon-build

```diff
@@ -0,0 +1,29 @@
+#!/bin/sh
+set -e
+
+private_storage_setup() {
+  [ -n "$HALCYON_AWS_ACCESS_KEY_ID" ] &&
+  [ -n "$HALCYON_AWS_SECRET_ACCESS_KEY" ] &&
+  [ -n "$HALCYON_S3_BUCKET" ]
+}
+
+[ -f ./.env ] && source ./.env
+
+if ! private_storage_setup; then
+  echo 'You must set the Halcyon variables for AWS credentials' >&2
+  echo 'https://halcyon.sh/reference/#private-storage-options' >&2
+  exit 1
+fi
+
+if [ ! -d /app/halcyon ]; then
+  echo 'Installing halcyon...'
+  git clone https://github.com/mietek/halcyon.git /app/halcyon
+fi
+
+# Note --no-clean-private-storage ensures developers don't need to rebuild their
+# stuff if another developer builds other versions for their own use (by
+# default, halcyon keeps only one version per platform at a time).
+/app/halcyon/halcyon build \
+  --no-clean-private-storage --cabal-version=1.22.0.0 "$@"
+
+ln -sf /app/sandbox/.halcyon-sandbox.config cabal.sandbox.config
```

Comment:
I’m surprised you have a symlink in `/app/sandbox`:
```
/app/sandbox/cabal.sandbox.config -> /app/sandbox/.halcyon-sandbox.config
```

I’d be expecting only the actual config in `/app/sandbox`, and the symlink in `/app`:
```
/app/cabal.sandbox.config -> /app/sandbox/.halcyon-sandbox.config
```

Can you list the contents of your sandbox directory archive?
```
tar -tf /var/tmp/halcyon-cache/halcyon-sandbox-fd49f80.701b05b-carnival-0.0.0.tar.gz
```